### PR TITLE
GH-36: Port ProducerListener from S-I-K

### LINKED ISSUE #36: Add ProducerListener to KafkaTemplate
Port from s-i-k

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves #36 (https://github.com/spring-projects/spring-kafka/issues/36)

### CODE REVIEW

File: spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java [hunk 1]
```diff
@@ -68,6 +73,15 @@ public void setDefaultTopic(String defaultTopic) {
 		this.defaultTopic = defaultTopic;
 	}
 
+	/**
+	 * Set a {@link ProducerListener} which will be invoked when kafka acknowledges
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java]: I'm sure Kafka should be capitalized in JavaDocs.

File: spring-kafka/src/main/java/org/springframework/kafka/support/DefaultProducerListener.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+/**
+ * No-op implementation of @link ProducerListener}, to be used as base class for other implementations.
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/support/DefaultProducerListener.java]: Missed left paren for `@link`

Comment (garyrussell) [inline spring-kafka/src/main/java/org/springframework/kafka/support/DefaultProducerListener.java]: Hehe - copy/paste from s-i-k

File: spring-kafka/src/main/java/org/springframework/kafka/support/DefaultProducerListener.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ */
+public class DefaultProducerListener<K, V> implements ProducerListener<K, V> {
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/support/DefaultProducerListener.java]: Typically in Spring similar classes are called like `...Adapter`. E.g. `ChannelInterceptorAdapter`

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/bcbf606fe4c8d1672e51c59dc7e192d88c09daf4

**Pull request closed without merging**